network/p2p/dnsaddr/resolveController.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (C) 2019-2023 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package dnsaddr
+
+import (
+	madns "github.com/multiformats/go-multiaddr-dns"
+
+	"github.com/algorand/go-algorand/tools/network"
+)
+
+// MultiaddrDNSResolveController returns a madns.Resolver, cycling through underlying net.Resolvers
+type MultiaddrDNSResolveController struct {
+	resolver      *madns.Resolver
+	nextResolvers []func() *madns.Resolver
+	controller    network.ResolveController
+}
```

Comment:
does this need to be exported when we have a constructor to create it?

Reply:
It doesn't right now, but since it's in the dnsaddr package I assume we will want to reference this type elsewhere, potentially without having to write an interface to obscure it.